api/clients/v2/validator_payload_retriever.go

```diff
@@ -0,0 +1,212 @@
+
+// BuildValidatorPayloadRetriever builds a ValidatorPayloadRetriever from config structs.
+func BuildValidatorPayloadRetriever(
+	logger logging.Logger,
+	validatorPayloadRetrieverConfig ValidatorPayloadRetrieverConfig,
+	ethConfig geth.EthClientConfig,
+	thegraphConfig thegraph.Config,
+	kzgConfig kzg.KzgConfig,
+) (*ValidatorPayloadRetriever, error) {
+	err := validatorPayloadRetrieverConfig.checkAndSetDefaults()
+	if err != nil {
+		return nil, fmt.Errorf("check and set config defaults: %w", err)
+	}
+
+	ethClient, err := geth.NewClient(ethConfig, gethcommon.Address{}, 0, logger)
+	if err != nil {
+		return nil, fmt.Errorf("new eth client: %w", err)
+	}
+
+	reader, err := eth.NewReader(
+		logger,
+		ethClient,
+		validatorPayloadRetrieverConfig.BlsOperatorStateRetrieverAddr,
+		validatorPayloadRetrieverConfig.EigenDAServiceManagerAddr)
+	if err != nil {
+		return nil, fmt.Errorf("new reader: %w", err)
+	}
+
+	chainState := eth.NewChainState(reader, ethClient)
+	indexedChainState := thegraph.MakeIndexedChainState(thegraphConfig, chainState, logger)
+
+	kzgVerifier, err := verifier.NewVerifier(&kzgConfig, nil)
+	if err != nil {
+		return nil, fmt.Errorf("new verifier: %w", err)
+	}
+
+	retrievalClient := NewRetrievalClient(
+		logger,
+		reader,
+		indexedChainState,
+		kzgVerifier,
+		int(validatorPayloadRetrieverConfig.ConnectionCount))
+
+	codec, err := codecs.CreateCodec(
+		validatorPayloadRetrieverConfig.PayloadPolynomialForm,
+		validatorPayloadRetrieverConfig.BlobEncodingVersion)
+	if err != nil {
+		return nil, fmt.Errorf("create codec: %w", err)
+	}
+
+	return &ValidatorPayloadRetriever{
+		logger:          logger,
+		config:          validatorPayloadRetrieverConfig,
+		codec:           codec,
+		retrievalClient: retrievalClient,
+		g1Srs:           kzgVerifier.Srs.G1,
+	}, nil
+}
+
+// NewValidatorPayloadRetriever creates a new ValidatorPayloadRetriever from already constructed objects
+func NewValidatorPayloadRetriever(
+	logger logging.Logger,
+	config ValidatorPayloadRetrieverConfig,
+	codec codecs.BlobCodec,
+	retrievalClient RetrievalClient,
+	g1Srs []bn254.G1Affine,
+) (*ValidatorPayloadRetriever, error) {
+	err := config.checkAndSetDefaults()
+	if err != nil {
+		return nil, fmt.Errorf("check and set config defaults: %w", err)
+	}
+
+	return &ValidatorPayloadRetriever{
+		logger:          logger,
+		config:          config,
+		codec:           codec,
+		retrievalClient: retrievalClient,
+		g1Srs:           g1Srs,
+	}, nil
+}
+
+// GetPayload iteratively attempts to retrieve a given blob from the quorums listed in the EigenDACert.
+//
+// If the blob is successfully retrieved, then the blob verified against the EigenDACert. If the verification succeeds,
+// the blob is decoded to yield the payload (the original user data, with no padding or any modification), and the
+// payload is returned.
+func (pr *ValidatorPayloadRetriever) GetPayload(
+	ctx context.Context,
+	eigenDACert *verification.EigenDACert,
```

Comment:
cert might need to be moved to some common/core package instead now that we also want to have a bindings version to be used for the contracts?

Reply:
I think that makes sense, but it's not clear to me exactly where it ought to be moved.

Reply:
yep lets. keep as is for now